src/main/java/seedu/address/logic/parser/AddEventCommandParser.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CATEGORY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ENDDATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ENDTIME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STARTDATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STARTTIME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.Set;
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.AddEventCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.common.Category;
+import seedu.address.model.common.Date;
+import seedu.address.model.common.Name;
+import seedu.address.model.common.Tag;
+import seedu.address.model.event.Event;
+import seedu.address.model.event.Time;
+
+/**
+ * Parses input arguments and creates a new AddEventCommand object
+ */
+public class AddEventCommandParser implements Parser<AddEventCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddEventCommand
+     * and returns an AddEventCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddEventCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_STARTDATE,
+                        PREFIX_STARTTIME, PREFIX_ENDDATE, PREFIX_ENDTIME, PREFIX_CATEGORY, PREFIX_TAG);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_STARTDATE, PREFIX_ENDDATE)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddEventCommand.MESSAGE_USAGE));
+        }
+
+        Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
+        Date startDate = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_STARTDATE).get());
+        Time startTime = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_STARTTIME).get());
+        Date endDate = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_ENDDATE).get());
+        Time endTime = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_ENDTIME).get());
+        Set<Category> categoryList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_CATEGORY));
```

Comment:
Maybe can change the parseTags here to a parseCategories? 🤔

Reply:
Thanks for spotting the error:)